src/libpriv/rpmostree-postprocess.c

```diff
@@ -1506,10 +1507,43 @@ rpmostree_treefile_postprocessing (GFile         *yumroot,
         }
     }
 
+  {
+    gboolean os_release_versioning = FALSE;
+
+    if (!_rpmostree_jsonutil_object_get_optional_boolean_member (treefile,
+                                                                 "os-release-versioning",
+                                                                 &os_release_versioning,
+                                                                 error))
+      goto out;
+
+    if (os_release_versioning && next_version != NULL)
+      {
+        g_autoptr(GFile) os_release = g_file_resolve_relative_path (yumroot, "etc/os-release");
+        /* check for existence while also following potential symlink */
+        if (g_file_query_file_type (os_release, 0, cancellable) == G_FILE_TYPE_REGULAR)
+          {
+            /* In theory we should be escaping the string using shell rules
+             * here... */
+            g_autofree char *contents =
+              g_strdup_printf ("\nOSTREE_VERSION=\"%s\"\n", next_version);
+            g_autoptr(GFileOutputStream) output =
+              g_file_append_to (os_release, 0, cancellable, error);
```

Comment:
Note in the future we will need to do a physical copy for this, so we might as well use `glnx_file_get_contents_utf8_at()` + `glnx_file_replace_contents_at()` or so.